Firestore/core/test/firebase/firestore/local/local_serializer_test.cc

```diff
@@ -353,6 +369,43 @@ TEST_F(LocalSerializerTest, EncodesQueryData) {
   ExpectRoundTrip(query_data, expected);
 }
 
+TEST_F(LocalSerializerTest, EncodesQueryDataWithDocumentQuery) {
+  core::Query query = Query("room/1");
+  TargetId target_id = 42;
+  ListenSequenceNumber sequence_number = 10;
+  SnapshotVersion version = testutil::Version(1039);
+  ByteString resume_token = testutil::ResumeToken(1039);
+
+  QueryData query_data(core::Query(query), target_id, sequence_number,
+                       QueryPurpose::Listen, SnapshotVersion(version),
+                       ByteString(resume_token));
+
+  // Let the RPC serializer test various permutations of query serialization.
+  ByteStringWriter writer;
+  google_firestore_v1_Target_DocumentsTarget proto =
+      remote_serializer.EncodeDocumentsTarget(query_data.query());
+  writer.WriteNanopbMessage(google_firestore_v1_Target_DocumentsTarget_fields,
+                            &proto);
+  remote_serializer.FreeNanopbMessage(
+      google_firestore_v1_Target_DocumentsTarget_fields, &proto);
+
+  ByteString query_target_bytes = writer.Release();
+  auto documents_target_proto =
```

Comment:
I think using our `Serializer` to create the representation here defeats the purpose of the check a little bit. Can you set up a `::firestore::client::Target::DocumentsTarget` manually instead and add it to `documents_proto`?